models/news-model.js

```diff
@@ -39,3 +39,37 @@ exports.countComments = () => {
       return Promise.all(commentCounts.rows);
     });
 };
+//merge 6 here
+
+exports.insertNewComment = (newComment, article_id) => {
+  if (
+    !Object.keys(newComment).includes("username") ||
+    !Object.keys(newComment).includes("body")
+  ) {
+    return Promise.reject({ status: 400, msg: "Error: 400 bad request" });
+  }
+
+  const { username, body } = newComment;
+  const checkingUserQuery = `SELECT username FROM users WHERE username = $1;`;
+
+  return db
+    .query(checkingUserQuery, [username])
+    .then((authorsArray) => {
+      if (authorsArray.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Error: 404 not found" });
+      }
+      return Promise.all(authorsArray.rows);
+    })
+    .then((authors) => {
+      const { username } = authors[0];
+      const query =
+        "INSERT INTO comments (body, author, article_id) VALUES ($1, $2, $3) RETURNING*;";
+      return db.query(query, [body, username, article_id]);
+    })
+    .then((post) => {
+      if (post.rows[0].length === 0) {
+        return Promise.reject({ status: 400, msg: "Error: 404 not found" });
```

Comment:
This error message will only be triggered if you haven't posted anything. Assuming your username check has been successful, and you have a body key, that shouldn't happen unless there is a PSQL error. That means this Promise.reject will never be triggered